projects/RabbitMQ.Client/client/impl/SessionBase.cs

```diff
@@ -135,14 +136,14 @@ public virtual void Transmit(in OutgoingCommand cmd)
             }
             // We used to transmit *inside* the lock to avoid interleaving
             // of frames within a channel.  But that is fixed in socket frame handler instead, so no need to lock.
-            cmd.Transmit(ChannelNumber, Connection);
+            Connection.Write(cmd.Transmit(ChannelNumber, Connection.FrameMax));
         }
 
-        public virtual void Transmit(IList<OutgoingCommand> cmds)
+        public virtual void Transmit<T>(IList<T> cmds) where T : struct, IOutgoingCommand
         {
             for (int i = 0; i < cmds.Count; i++)
             {
-                cmds[i].Transmit(ChannelNumber, Connection);
+                Connection.Write(cmds[i].Transmit(ChannelNumber, Connection.FrameMax));
```

Comment:
Probably not going to make a huge difference but technically we don't need to reacquire the FrameMax inside the loop all the time right?

Reply:
Good catch.